src/compiler/builder.ts

```diff
@@ -614,10 +765,24 @@ namespace ts {
     export interface BuilderProgram {
         /*@internal*/
         getState(): BuilderProgramState;
+        /*@internal*/
+        backupCurrentState(): void;

```

Comment:
minor nit (up to you if you want to change this): Would `backupState`/`restoreState` be clearer?